### PR TITLE
[do not merge, need rebase] (planner 2) add planning variables to plannerShared

### PR DESCRIPTION
**Pull request by janiversen**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

move handling of planning variables from desktop-widgets to plannerShared, to prepare for an identical use in QML.

creating the functions in plannerShared and using the new functions in desktop-widget are on purpose in separate commits.

creating the new functions are grouped in commits with similar functionality

** Warning  ** this pr is based on #2464 and thus contains those commits

### CODE REVIEW

File: backend-shared/plannershared.cpp [hunk 1]
```diff
@@ -113,3 +113,13 @@ void plannerShared::set_vpmb_conservatism(int value)
...
 }
+
+bool plannerShared::dobailout()
+{
+	return qPrefDivePlanner::dobailout();
+}
+void plannerShared::set_dobailout(bool value)
+{
+	qPrefDivePlanner::set_dobailout(value);
+	DivePlannerPointsModel::instance()->emitDataChanged();
```
Comment (dirkhh) [inline backend-shared/plannershared.cpp:124]: so why is this done differently from the cases in the previous commit, where you have this asymmetric pattern of reading from qPref and calling the setter in `DivePlannerPointsModel` ?
Calling that signal in the unrelated class seems odd. But maybe there's a good reason to do it this way that I just don't see?

**Review by dirkhh — COMMENTED** [review]
one small comment about something that looks like an inconsistency to me.
overall I'd prefer a symmetric setup where we do both setters and getters through the same class, even if that causes an extra function call / class dereference.

File: backend-shared/plannershared.cpp (hunk 1 above)
Comment (janiversen) [reply backend-shared/plannershared.cpp:124]: I too find it odd, but desktop-widgets/diveplanner.cpp have a number of case like this:

before:
void PlannerSettingsWidget::setBackgasBreaks(bool dobreaks)
 {
       qPrefDivePlanner::instance()->set_doo2breaks(dobreaks);
       plannerModel->emitDataChanged();
 }
after:
void PlannerSettingsWidget::setBackgasBreaks(bool dobreaks)
 {
       plannerShared::set_doo2breaks(dobreaks);
 }

Where you see I move the functionality 1-1 to backend-shared.

Actually I do not know if there is a good reason.

Comment (dirkhh) [reply backend-shared/plannershared.cpp:124]: That's a good explanation. @atdotde @bstoeger - any insights into why we do things that way?
(I think Berthold is offline, now that I wrote this)

**Comment by dirkhh** [comment]
manually merged

**Pull request closed without merging**